src/nvim/os/fs.c

```diff
@@ -279,13 +279,24 @@ bool os_file_is_readonly(const char *name)
   return access(name, W_OK) != 0;
 }
 
+/// Check if a file is readable.
+///
+/// @return 0 if `name` is not writable,
+/// @return 1 if `name` is writable,
+/// @return 2 for a directory which we have rights to write into.
+bool os_file_is_readable(const char *name)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_WARN_UNUSED_RESULT
+{
+  return (access(name, R_OK) == 0);
```

Comment:
Should be ok, just need to add a define for R_OK to 04 ([docs](https://msdn.microsoft.com/en-us/library/1w06ktdy.aspx))